src/solver/where/backward_where.cpp

```diff
@@ -0,0 +1,283 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "miopen/kernel_info.hpp"
+#include "miopen/mlo_internal.hpp"
+#include "miopen/tensor.hpp"
+#include "miopen/tensor_view_utils.hpp"
+#include "miopen/where/problem_description.hpp"
+#include <cstddef>
+#include <miopen/datatype.hpp>
+#include <miopen/kernel_build_params.hpp>
+#include <miopen/where/invoke_params.hpp>
+#include <miopen/where/solvers.hpp>
+#include <miopen/where.hpp>
+#include <miopen/target_properties.hpp>
+#include "../kernels/tensor_view.hpp"
+
+#define LOCAL_SIZE 256
+
+namespace miopen {
+
+namespace solver {
+
+namespace where {
+
+template <int N>
+int64_t check_broadcasted_contiguous(const tensor_view_t<N>& tensorView)
+{
+    int64_t num_elems = 1;
+
+    for(int i = N - 1; i >= 0; i--)
+    {
+        if(tensorView.stride[i] != 0 && tensorView.stride[i] != num_elems)
+            return 0;
+        if(tensorView.stride[i] == 0)
+        {
+            for(int j = i; j >= 0; j--)
+                if(tensorView.stride[j] != 0)
+                    return 0;
+            return num_elems;
+        }
+        num_elems *= tensorView.size[i];
+    }
+
+    return num_elems;
+}
+
+bool WhereBackward::IsApplicable(const ExecutionContext& context,
+                                 const miopen::where::BackwardProblemDescription& problem) const
+{
+    std::ignore                     = context;
+    tensor_view_t<5> input_grad_tv  = get_inner_expanded_tv<5>(problem.GetInputGradDesc());
+    tensor_view_t<5> other_grad_tv  = get_inner_expanded_tv<5>(problem.GetOtherGradDesc());
+    tensor_view_t<5> cond_tv        = get_inner_expanded_tv<5>(problem.GetConditionDesc());
+    tensor_view_t<5> output_grad_tv = get_inner_expanded_tv<5>(problem.GetOutputGradDesc());
+
+    input_grad_tv = broadcast_to(input_grad_tv, output_grad_tv);
+    other_grad_tv = broadcast_to(other_grad_tv, output_grad_tv);
+    cond_tv       = broadcast_to(cond_tv, output_grad_tv);
+
+    auto cond_contig_size        = check_broadcasted_contiguous(cond_tv);
+    auto input_grad_contig_size  = check_broadcasted_contiguous(input_grad_tv);
+    auto other_grad_contig_size  = check_broadcasted_contiguous(other_grad_tv);
+    auto output_grad_contig_size = check_broadcasted_contiguous(output_grad_tv);
+
+    auto is_all_contiguous =
+        isTensorViewContiguous(input_grad_tv) && isTensorViewContiguous(other_grad_tv) &&
+        isTensorViewContiguous(cond_tv) && isTensorViewContiguous(output_grad_tv);
+
+    bool is_all_broadcasted_contiguous = cond_contig_size && output_grad_contig_size &&
```

Comment:
There are similar codes in three places.
Make duplicate code a separate functions.